src/app/shared/entry/descriptor-language.service.ts

```diff
@@ -14,28 +14,71 @@
+import { ToolDescriptor } from '../swagger';
 import { MetadataService } from './../swagger/api/metadata.service';
 import { DescriptorLanguageBean } from './../swagger/model/descriptorLanguageBean';
-import { ToolDescriptor } from '../swagger';
 
 @Injectable()
 export class DescriptorLanguageService {
-  public descriptorLanguages$: Observable<Array<ToolDescriptor.TypeEnum>>;
+  private descriptorLanguages$: Observable<Array<ToolDescriptor.TypeEnum>>;
   private descriptorLanguagesBean$ = new BehaviorSubject<DescriptorLanguageBean[]>([]);
-  constructor(private metadataService: MetadataService) {
-      this.update();
-      this.descriptorLanguages$ = this.descriptorLanguagesBean$.pipe(map(descriptorLanguageMap => {
-          if (descriptorLanguageMap) {
-              return descriptorLanguageMap.map((descriptorLanguage) => <ToolDescriptor.TypeEnum>descriptorLanguage.value.toString());
-          }
-      }));
+  public filteredDescriptorLanguages$: Observable<Array<ToolDescriptor.TypeEnum>>;
+
+  constructor(private metadataService: MetadataService, private sessionQuery: SessionQuery) {
+    this.update();
+    this.descriptorLanguages$ = this.descriptorLanguagesBean$.pipe(
+      map(descriptorLanguageMap => {
+        if (descriptorLanguageMap) {
+          return descriptorLanguageMap.map(descriptorLanguage =>
+            this.convertDescriptorLanguageBeanToToolDescriptorTypeEnum(descriptorLanguage)
+          );
+        }
+      })
+    );
+    const combined$ = combineLatest(this.descriptorLanguages$, this.sessionQuery.entryType$);
+    this.filteredDescriptorLanguages$ = combined$.pipe(
+      map(combined => this.filterLanguages(combined[0], combined[1]))
+    );
   }
   update() {
     this.metadataService.getDescriptorLanguages().subscribe((languageBeans: Array<DescriptorLanguageBean>) => {
       this.descriptorLanguagesBean$.next(languageBeans);
     });
   }
+
+  /**
+   * In almost all cases, the string value from the DescriptorLanguageBean matches the ToolDescriptor.TypeEnum exactly.
+   * However, for services, it does not.
+   *
+   * @param {DescriptorLanguageBean} descriptorLanguageBean
+   * @returns
+   * @memberof DescriptorLanguageService
+   */
+  convertDescriptorLanguageBeanToToolDescriptorTypeEnum(descriptorLanguageBean: DescriptorLanguageBean): ToolDescriptor.TypeEnum {
+    if (descriptorLanguageBean.value === 'service') {
+      return ToolDescriptor.TypeEnum.DOCKSTORESERVICE;
+    } else {
+      return <ToolDescriptor.TypeEnum>descriptorLanguageBean.value.toString();
+    }
+  }
+
+  /**
+   * Some entries are not meant to show all descriptor types
+   *
+   * @param {ToolDescriptor.TypeEnum[]} thing
+   * @param {EntryType} entryType
+   * @returns {ToolDescriptor.TypeEnum[]}
+   * @memberof DescriptorLanguageService
+   */
+  filterLanguages(thing: ToolDescriptor.TypeEnum[], entryType: EntryType): ToolDescriptor.TypeEnum[] {
+    if (entryType === EntryType.BioWorkflow || EntryType.Tool || !entryType) {
```

Comment:
Shouldn't this be `entryType === EntryType.Bioworkflow || entryType === EntryType.Tool || !entryType`?